src/main/java/com/exactpro/th2/common/event/Event.java

```diff
@@ -266,6 +275,32 @@ public Event messageID(MessageID attachedMessageID) {
         return eventBuilder.build();
     }
 
+    public List<EventBatch> toListBatchProto(@Nullable EventID parentID) throws JsonProcessingException {
+        List<com.exactpro.th2.common.grpc.Event> events = toListProto(parentID);
+
+        EventBatch.Builder builder = EventBatch.newBuilder()
+            .addAllEvents(events);
+
+        if (events.size() != 1) {
+            builder.setParentEventId(parentID);
+        }
+
+        return Collections.singletonList(builder.build());
+    }
+
+    public List<EventBatch> toListBatchProto(int maxEventBatchContentSize, @Nullable EventID parentID) throws JsonProcessingException {
+        if (maxEventBatchContentSize <= 0) {
+            throw new IllegalArgumentException("'maxEventBatchContentSize' should be greater than zero, actual: " + maxEventBatchContentSize);
+        }
+
+        List<com.exactpro.th2.common.grpc.Event> events = toListProto(parentID);
+
+        List<EventBatch> result = new ArrayList<>();
+        Map<EventID, List<com.exactpro.th2.common.grpc.Event>> eventGroups = events.stream().collect(Collectors.groupingBy(com.exactpro.th2.common.grpc.Event::getParentId));
+        batch(maxEventBatchContentSize, result, eventGroups, parentID);
```

Comment:
What will happen if the `null` parentID was passed? As far as I remember the default value for Event::getParentId will be used. So when we will call `batch` with null `parentID` parameter the exception will be thrown because `eventGroups` does not have group for null parent ID